packages/peregrine/lib/talons/CheckoutPage/__tests__/useCheckoutPage.spec.js

```diff
@@ -0,0 +1,358 @@
+            { cartId: '123' },
+            { createCart: () => {}, removeCart: () => {} }
+        ]);
+
+        const { talonProps } = getTalonProps(props);
+
+        await talonProps.handlePlaceOrder();
+
+        expect(getOrderDetails).toHaveBeenCalledWith({
+            variables: { cartId: '123' }
+        });
+        expect(placeOrder).toHaveBeenCalledWith({
+            variables: { cartId: '123' }
+        });
+    });
+
+    test('should remove and create new cart', async () => {
+        const createCart = jest.fn();
+        const removeCart = jest.fn();
+        const fetchCartId = jest.fn();
+        useCartContext.mockReturnValueOnce([
+            { cartId: '123' },
+            { createCart, removeCart }
+        ]);
+        createCartMutationResult.mockReturnValue([fetchCartId]);
+
+        const { talonProps } = getTalonProps(props);
+
+        await talonProps.handlePlaceOrder();
+
+        expect(removeCart).toHaveBeenCalled();
+        expect(createCart).toHaveBeenCalledWith({ fetchCartId });
+    });
+
+    test('should clear cart data from cache', async () => {
+        const { talonProps } = getTalonProps(props);
+
+        await talonProps.handlePlaceOrder();
+
+        expect(clearCartDataFromCache).toHaveBeenCalledWith(client);
+    });
+});
+
+test('hasError should be true if place order mutation failed with errors', () => {
+    placeOrderMutationResult.mockReturnValueOnce([
+        () => {},
+        { data: null, loading: false, error: 'some error' }
+    ]);
+
+    const { talonProps } = getTalonProps(props);
+
+    expect(talonProps.hasError).toBeTruthy();
+});
+
+describe('isCartEmpty', () => {
+    test('should be true if checkout data is falsy', () => {
+        getCheckoutDetailsQueryResult.mockReturnValueOnce({
+            data: null,
+            error: null,
+            loading: false
+        });
+
+        const { talonProps } = getTalonProps(props);
+
+        expect(talonProps.isCartEmpty).toBeTruthy();
+    });
+
+    test('should be true if total quantity is 0', () => {
+        getCheckoutDetailsQueryResult.mockReturnValueOnce({
+            data: {
+                cart: {
+                    total_quantity: 0
+                }
+            },
+            error: null,
+            loading: false
+        });
+
+        const { talonProps } = getTalonProps(props);
+
+        expect(talonProps.isCartEmpty).toBeTruthy();
+    });
+
+    test('should be false if total quantity is not 0', () => {
+        getCheckoutDetailsQueryResult.mockReturnValueOnce({
+            data: {
+                cart: {
+                    total_quantity: 5
+                }
+            },
+            error: null,
+            loading: false
+        });
+
+        const { talonProps } = getTalonProps(props);
+
+        expect(talonProps.isCartEmpty).toBeFalsy();
+    });
+});
+
+test('isGuestCheckout should be negation of isSignedIn from useUserContext', () => {
```

Comment:
Careful of tests like this, in my opinion this is a little too implementation-specific and therefore brittle.

Reply:
@supernova-at this is a good point and leads me to believe we should drop the `isGuestCheckout` value and  just be returning `isSignedIn` for use in the UI component.

This raises a question for dev time though. In most of our talons we return some simple boolean states such as loading or render flags such as `isSignedIn`. How should we test those?

Reply:
I wanted to test the fact that it is dependent on `isSignedIn` which is from another talon. I wanted to do this because, if someone changes the logic, the test has to fail.

All this is based on my assumptions about testing such booleans. 

@sirugh nice idea. Good topic to discuss.